src/tools/Erase.js

```diff
@@ -35,40 +33,80 @@ const _Tool = types
       return <ToolView item={self} />;
     },
   }))
-  .actions(self => ({
-    updateCursor() {
-      const val = 24;
-      const stage = self.obj.stageRef;
-      const base64 = Canvas.brushSizeCircle(val);
-      const cursor = ["url('", base64, "')", " ", Math.floor(val / 2) + 4, " ", Math.floor(val / 2) + 4, ", auto"];
-
-      stage.container().style.cursor = cursor.join("");
-    },
+  .actions(self => {
+    let pathPoints;
+    let brush;
+    return {
+      afterCreate() {
+        const dispose = reaction(
```

Comment:
here is the same. this observer looks scary and monstrous, even if that's a bit more correct. Brush/Erase buttons should be reworked a bit, as well as their behaviour. But for now there is only one change point as I know — `onClick`

Reply:
I'd say change point is `setSelected` in this case. And here I removed mobx reactions too.